tapdb/assets_store.go

```diff
@@ -3167,17 +3167,22 @@ func (a *AssetStore) LogAnchorTxConfirm(ctx context.Context,
 			isBurn := !isNumsKey && len(witnessData) > 0 &&
 				asset.IsBurnKey(scriptPubKey, witnessData[0])
 			isKnown := fullScriptKey.Type != asset.ScriptKeyUnknown
+			isRemotePedersen := fullScriptKey.Type ==
+				asset.ScriptKeyUniquePedersen &&
+				!out.ScriptKeyLocal
 			skipAssetCreation := !isTombstone && !isBurn &&
-				!out.ScriptKeyLocal && !isKnown
+				!out.ScriptKeyLocal &&
+				(!isKnown || isRemotePedersen)
```

Comment:
Could use a comment here re the diff re handling pederson types vs all the rest.

Reply:
Yeah, good point. I did that and also extracted the logic into a standalone function and added unit tests for it: 20fa3815